### ISSUE TITLE
Documentation for grammar tagged nested syntax is unclear.

### ISSUE THREAD

**Issue by ryceg** [incobalt/Trice#2]
It is unclear how to format nested grammars in passages. An example of nested grammars (inc. grammars that call other grammars) would be greatly appreciated!

**Comment by incobalt** [comment]
If you mean where grammar passages can link to other passages to make more complex generations, I mention this in my documentation here: https://github.com/incobalt/Trice#building-a-grammar-with-twine-passages and TriceExample shows how to do this in the gHabitatFeature grammar passage. In what way is the documentation unlclear or the example insufficient? I'd like to know so that I can make either one better for people!

**Comment by ryceg** [comment]
Oh! My mistake. Apologies, poor reading comprehension, at least for that part. I'm still unsure how to implement grammars in arrays.

```
:: pie [grammar]
pumpkin
chocolate
apple and ["pear", "blackberry"]
pecan
```

Here, whenever the apple line is selected, I'm trying to achieve the output of 
```
apple and pear
apple and blackberry
```
equally often. What's the formatting for that? And what about arrays containing more grammars? i.e. if instead of pear and blackberry it was #fruit# and #berry#, each two passages containing different fruits and berries respectively.

**Comment by incobalt** [comment]
Oh, I see. Tracery can't quite do this as far as I know. It can do something similar by storing a value and then later expanding that value like this:
`apple and [filling:pear, blackberry, #fruit#, #berry#]#showFilling#`
and then the showFilling passage is just:
`#filling#`
This would set the value of filling to a selection in the list with equal weighting, and then you have to call it from a different rule because of the way Tracery expands rules.

**However this isn't how I'd do it with Trice**
I would instead make a grammar passage named filling that reads:
```
[[fruit]]
[[berry]]
```
and in your pie passage I'd write:
`apple and [[filling]]`
(and I'd fill out a passage for fruit and one for berry with their respective fillings)
This will generate links between the passages in Twine. Of course if you don't want the arrows, you can use Tracery syntax like #fruit# and #berry# instead.
Basically, each grammar passage is an array. There's no elegant way of doing it inline in Tracery, as far as I know.

**Comment by mrfb** [comment]
You might also try something like this, which is a trick I use often—especially if I wouldn’t use the extra grammars anywhere else:

```
:: pie [grammar]
pumpkin
pumpkin
chocolate
chocolate
apple and pear
apple and blackberry
pecan
pecan
```

**Comment by incobalt** [comment]
This is probably the most elegant solution to add once-off rules that you don't want to make specific passages for. I admit it would be tedious to make a passage just to choose between two options a single time. Beyond that, you might need to extend Tracery to have a modifier that will flatten an inline rule, but I'm talking out of my depth here.
